examples/fairmotion_interface.py

```diff
@@ -150,37 +143,76 @@ def metadata_parser(self, metadata_dict: Dict[str, Any], to_file: bool):
                     data[k] = mn.Vector3(v)
         return data
 
-    def save_metadata(self, name: str):
+    def save_metadata(self, file: str):
         """
         Saves the current metadata to a json file in given file path
         """
-        filepath = f"{self.metadata_dir}{name + FILE_SUFFIX}.json"
-        data = {name: self.metadata_parser(self.metadata[name], to_file=True)}
+        meta_access = [METADATA_DIR + file + ".json", METADATA_DIR + file, file]
+
+        if file:
+            # sanitizing file path
+            for filename in meta_access:
+                if os.path.exists(filename):
+                    file = filename
+                    break
+
+                # we are no longer overwriting a file
+                elif "/" not in file:
+                    # file is not a file path, we need to aim it at our directory
+                    if ".json" not in file:
+                        # add file type
+                        file = file + ".json"
+                    file = METADATA_DIR + file
+        else:
+            # generate filename from timestamp
+            file = METADATA_DIR + "user_" + time.strftime("%Y-%m-%d_%H-%M-%S")
+
+        logger.info(f"Saving data to file: {file}")
+
+        # updating user_metadata to reflect characters position
+        if self.rotation_offset and self.translation_offset:
+            data = self.user_metadata
+            data["rotation"] = self.rotation_offset
+            data["translation"] = self.translation_offset
 
-        with open(filepath, "w") as file:
-            json.dump(data, file)
+        data = self.metadata_parser(self.user_metadata, to_file=True)
 
-    def fetch_metadata(self, name):
+        with open(file, "w") as f:
+            json.dump(data, f)
+        logger.info(f"Saved: {file}")
+
+    def fetch_metadata(self, file):
         """
-        Fetch metadata from a json file in given file name and sets current metadata
+        Fetch metadata from a json file in given file name and sets metadata
         """
-        if FILE_SUFFIX in name and len(name) > len(FILE_SUFFIX):
-            # if name give is full name of file, truncate FILE_SUFFIX
-            filename = name
-            name = name[: -len(FILE_SUFFIX)]
-        else:
-            # including FILE_SUFFIX if not included in name
-            filename = name + FILE_SUFFIX
+        meta_access = [METADATA_DIR + file + ".json", METADATA_DIR + file, file]
 
-        filepath = f"{self.metadata_dir}{filename}.json"
+        # sanitizing file path
+        for filename in meta_access:
+            if os.path.exists(filename):
+                file = filename
+                break
 
         try:
-            with open(filepath, "r") as file:
-                data = json.load(file)
-        except Exception:
-            raise Exception("Error: File does not appear to exist.")
+            with open(file, "r") as f:
+                data = json.load(f)
+        except FileNotFoundError:
```

Comment:
We are you just reraising the error? Shouldn't we juse remove the try except entirely?